### PR TITLE
Allow OffscreenCanvas to work as CanvasResource

### PR DESCRIPTION
**Pull request by mudcube**
##### Description of change
Allows OffscreenCanvas to be used as a source for CanvasResource.

The code references OffscreenCanvas with the `global.` prefix, as some browsers do not yet support this feature. Also, `source` is checked to be defined for the same reason. There's a few ways this could be setup, but this looked the easiest to read to me. Updated JSDoc to match changes.

##### Pre-Merge Checklist
- [x] Documentation is changed or added.

### CODE REVIEW

**Comment by bigtimebuddy** [comment]
Reference issue #4538.

Thanks for the PR, @mudcube.

There's a linting problem, if could please fix that. Also, at the very least we need an example to verify that this works. Would you mind putting something together? One better would be to create an automated test that makes sure a resource can be created from an OffscreenCanvas. Our test environment should support this.

**Comment by mudcube** [comment]
Hi @bigtimebuddy, my apologies for overlooking those points, the latest revision should resolve the linting issue. Here's a test that currently fails, but works with the modifications in this pull request: https://www.pixiplayground.com/#/edit/Z5fNHO7iKnHhOsOS~Q956

**Comment by ivanpopelyshev** [comment]
I've modified your example.

OK, we cant test PR's from other repos, only through hacks.

Can you make it that way it actually shows something with that texture?

**Comment by rjenk88** [comment]
High gestures are most sequence to supurb nest creating a vision at posting noble Forbe based featuring motive s

**Comment by mudcube** [comment]
@ivanpopelyshev I've updated the example to load the texture into a PIXI.Sprite and display that onto screen, so you can visually see the modification sorts out the issue.

File: packages/core/src/textures/resources/CanvasResource.js [hunk 1]
```diff
@@ -13,11 +13,14 @@ export default class CanvasResource extends BaseImageResource
...
-     * @param {*} source - The source object
+     * @param {HTMLCanvasElement|OffscreenCanvas} source - The source object
      * @return {boolean} `true` if <canvas>
      */
     static test(source)
     {
-        return (source instanceof HTMLCanvasElement);
+        return (source && (
+            source instanceof HTMLCanvasElement
+            || source instanceof global.OffscreenCanvas
```
Comment (themoonrat) [inline packages/core/src/textures/resources/CanvasResource.js]: the 'global' namespace may not exist.  Shouldn't it work as just `source instanceof OffscreenCanvas`?

Comment (mudcube) [reply packages/core/src/textures/resources/CanvasResource.js]: > the 'global' namespace may not exist

Ah, yes, I see that `global` didn't make it into the final specs... Do you think `globalThis` or `window` would be a better option?

Comment (bigtimebuddy) [reply packages/core/src/textures/resources/CanvasResource.js]: `window` works, we use that in other places

**Comment by bigtimebuddy** [comment]
Awesome, great work 🎉

**Comment by JetLua** [comment]
#5780

**Pull request merged**